zircon.core/common/src/main/kotlin/org/hexworks/zircon/api/util/BSPTree.kt

```diff
@@ -0,0 +1,103 @@
+package org.hexworks.zircon.api.util
+
+import org.hexworks.zircon.api.data.Position
+import org.hexworks.zircon.api.data.Rect
+import org.hexworks.zircon.api.data.Size
+import kotlin.jvm.JvmStatic
+import kotlin.random.Random
+
+class BSPTree(var parent: BSPTree?, rec: Rect) {
+
+    companion object {
+        @JvmStatic
+        var minSize = 12
+
+        fun toMatrix(array: Array<CharArray>, BSPTrees: MutableList<BSPTree?>) {
+            var nbr = 48
+
+            for(BSPTree: BSPTree? in BSPTrees) {
+                if(BSPTree != null) {
+
+                    var char = nbr.toChar()
+                    var rec = BSPTree.room
+                    println(char + " " +BSPTree.boundingBox)
+                    if(rec != null) {
```

Comment:
You can have a function to check for presence of room like this:

```kotlin
fun whenHasRoom(fn: (Rect) -> Unit) {
    room.map(fn)
}
```

Use it like this instead of the `null` check:

```kotlin
BSPTree.whenHasRoom { rec ->
// ...
```